### PR TITLE
HTPP : replace decode with cpp-netlib decoded 

### PR DESCRIPTION
**Pull request by guzzijones**
---
**By submitting this pull-request, I confirm the following:**

- I have read and understood the [Contributing Guide](https://github.com/monero-project/kovri/blob/master/doc/CONTRIBUTING.md).
- I have checked that another pull-request for this purpose does not exist.
- I have considered and confirmed that this submission will be valuable to others.
- I accept that this submission may not be used and that this pull-request may be closed by the will of the maintainer.
- I give this submission freely under the BSD 3-clause license.
---

### CODE REVIEW

**Comment by guzzijones** [comment]
I need to double check the jumpservice.  This PR is to start a conversation so things don't drag out if the jump service leads down a rabbit hole.

**Comment by guzzijones** [comment]
I tested with this site which has plenty of % characters to decode.

`http://equationsecurity.i2p/?i2paddresshelper=XoJuBcTF8PW713nqoP0niK3CahOujCT~fR5rTqBd~c-PrSn9xgIZFnxUT7M9VMR6GGKpkDRNTZ3HOBTbG3kF3F2cFfV73UJOufh1Ok3ogMvs3tg~NLE9x7e9xbLfj2ckJTbiXoQkr80YpprT1h8fG-56u~8cqAwC7YRyZFAqHAK6m19NCe1X1yJDtKRdE-84za76wNXSkyJyco9Bd3Gk7sA0VwD5GGLYVVB-IurDIvUajkNtWFbKYj8Qmts~U2z8e9TvMMzQO5nnuXOsiZvOcLkkdtJRTgjoHq0Bnde8CWN44Y~u4fuBDmefbFBiq6Bp5Oik7h0JGqVXO~865H4g1We~DrGD7kRCQ8ziWlgcC3BzpOFh7FJ1oZFpNWSn0EPCOT7HOlbHgSsTdiB~CbHMNXnFrS-HD8MCgnT4NkklnMf3xjXm~S7H1XUV~VlumI1b47n6V10fcMMr77jfesIsd2B41HM4TFZiVhIJJZFlFOm5a1EqF62wM5zgUXEs-3oNBQAEAAcAAA%3d%3d#!sig%3dzJmi97pV9d446rnBKUl1gTkJhTNxEVkk6cLxS0DKBZU6yranF6nlxjrhVdJLPc2MaFfoxd2TtMEAOyeym0oBDA%3d%3d`

**Comment by guzzijones** [comment]
I made need another build run here.  It compiles just fine on my end after a make clean.

**Comment by guzzijones** [comment]
i may need a refresh on the build here. it compiles fine on my end and the tests run fine.

File: src/client/proxy/http.cc [hunk 1]
```diff
@@ -36,6 +36,8 @@
 #include <boost/algorithm/string.hpp>
 #include <boost/algorithm/string/regex.hpp>
 
+
```
Comment (anonimal) [inline src/client/proxy/http.cc]: Watch for these.

**Pull request merged**